### DISCUSSION TITLE
Dynamic params for Router

### DISCUSSION THREAD

**Discussion by Vince-vegas** [vercel/next.js#17449] [Help] [answered]
How can I pass any **params** on a route without implementing more params in **getStaticPaths**

e.g 

```
export async function getStaticPaths() {
  return {
    paths: [
      {
        params: {
          id: '1',
        },
      },
      {
        params: {
          id: '720',
        },
      },
      // **to any other params**
    ],
    fallback: false,
  };
}
```

Is there any solution that I could implement any **id parameters** in just a single **params**.

**Comment by IRediTOTO** [comment, marked as answer]
My experience is return nothing :D 
```
export async function getStaticPaths() {
  return {
    paths: [    ],
    fallback: true,
  };
}
```
Just that, this save tons of build time :D `fallback: true,` will do the rest 
if you still want? Use array then map it.
You will need query list of post ID in database something like this



export async function getStaticPaths() {

const listID=queryDatabaseFunc.findIDs()  // return [1,4,5,6,7,9]
const myListParam= listID.map(id=>{

return {params:{id}}
})


  return {
    paths: myListParam;
}

**Comment by Vince-vegas** [comment]
Awh getInitialProps solves it
